src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionResolveEndpoint.cs

```diff
@@ -174,22 +168,15 @@ public Task<CompletionItem> Handle(CompletionItem completionItem, CancellationTo
 
             if (tagHelperMarkupTooltip != null)
             {
-                var documentation = new StringOrMarkupContent(tagHelperMarkupTooltip);
-                completionItem = completionItem with { Documentation = documentation };
+                completionItem.Documentation = tagHelperMarkupTooltip;
             }
 
-            // We might strip out the commitcharacters for speed, bring them back
```

Comment:
Because we're deserializing directly into a completion item with `VsCommitCharacters` now we never lose the data so we don't need to restore commit characters anymore